flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/scalar/JsonUnquoteFunction.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.runtime.functions.scalar;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.table.data.binary.BinaryStringData;
+import org.apache.flink.table.functions.BuiltInFunctionDefinitions;
+import org.apache.flink.table.functions.SpecializedFunction.SpecializedContext;
+import org.apache.flink.table.runtime.functions.SqlJsonUtils;
+
+import javax.annotation.Nullable;
+
+/** Implementation of {@link BuiltInFunctionDefinitions#JSON_UNQUOTE}. */
+@Internal
+public class JsonUnquoteFunction extends BuiltInScalarFunction {
+
+    public JsonUnquoteFunction(SpecializedContext context) {
+        super(BuiltInFunctionDefinitions.JSON_UNQUOTE, context);
+    }
+
+    private static boolean isValidJsonVal(String jsonInString) {
+        // See also BuiltInMethods.scala, IS_JSON_VALUE
+        return SqlJsonUtils.isJsonValue(jsonInString);
+    }
+
+    private String unquote(String inputStr) {
+
+        StringBuilder result = new StringBuilder();
+        int i = 1;
```

Comment:
we are skipping the first and the last character without checking whether they are double quotes.
The string `[1, 2, 3]` is valid JSON and we return `1, 2, 3`.
I think we also need to check if the input string is surrounded by double quotes (even before checking if it's valid JSON) and return the input string if the quotes are missing.

For reference, this is MySQL's output:
```
mysql> select json_unquote('[1, 2, 3]');
+---------------------------+
| json_unquote('[1, 2, 3]') |
+---------------------------+
| [1, 2, 3]                 |
+---------------------------+
1 row in set (0.00 sec)

mysql> select json_unquote('"[1, 2, 3]"');
+-----------------------------+
| json_unquote('"[1, 2, 3]"') |
+-----------------------------+
| [1, 2, 3]                   |
+-----------------------------+
1 row in set (0.00 sec)
```

These cases should also be covered by tests.

Reply:
Great catch @fhueske 👍, I completely missed handling this.